src/ast/nodes/ImportExpression.ts

```diff
@@ -296,13 +308,41 @@ export default class ImportExpression extends NodeBase {
 			interop
 		}: NormalizedOutputOptions,
 		{ _, getDirectReturnFunction, getDirectReturnIifeLeft }: GenerateCodeSnippets,
-		pluginDriver: PluginDriver
+		pluginDriver: PluginDriver,
+		ownChunk: Chunk,
+		targetChunk: Chunk | null
 	): { helper: string | null; mechanism: DynamicImportMechanism | null } {
+		function getChunkInfoWithPath(chunk: Chunk): PreRenderedChunkWithFilename {
+			return { preliminaryFilename: chunk.getFileName(), ...chunk.getPreRenderedChunkInfo() };
+		}
```

Comment:
It looks to me like we can extract this function to the top level to avoid recreating it on every call?